lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -376,7 +377,9 @@ prefilterBlock b u0 = runState $ do
         ourU <- state $ utxoOurs tx
         let ourIns = Set.fromList (inputs tx) `Set.intersection` dom (u <> ourU)
         let u' = (u <> ourU) `excluding` ourIns
-        let received = fromIntegral @_ @Integer $ balance ourU
+        let withdrawalAmt =
+                sum $ map (fromIntegral . getCoin) $ Map.elems $ withdrawals tx
+        let received = fromIntegral @_ @Integer $ (withdrawalAmt + balance ourU)
         let spent = fromIntegral @_ @Integer $ balance (u `restrictedBy` ourIns)
```

Comment:
So actually, the withdrawals count towards the `spent` here (it's like an input that has been spent!)